src/histogram.c

```diff
@@ -5,16 +5,43 @@
 #include "nodes/makefuncs.h"
 #include "utils/lsyscache.h"
 #include <netinet/in.h>
+#include <lib/stringinfo.h>
+#include <libpq/pqformat.h>
 
 #include "compat.h"
 
+#define SUBAGG_AVG_NAME "avg"
+#define SUBAGG_COUNT_NAME "count"
+#define SUBAGG_MIN_NAME "min"
+#define SUBAGG_MAX_NAME "max"
+#define SUBAGG_SUM_NAME "sum"
+
+typedef enum SubAgg
+{
+	SUBAGG_AVG = 0,
+	SUBAGG_COUNT,
+	SUBAGG_MIN,
+	SUBAGG_MAX,
+	SUBAGG_SUM,
+	_MAX_SUBAGG,
+} SubAgg;
+
+static const char *subagg_names[_MAX_SUBAGG + 1] = {
```

Comment:
Rather than hard-coding the supported aggregate functions, can't we do this for the general case. Internally, I believe we should be able to check whether a given function `OID` is an aggregate function with the right input and output. Look at, e.g., `parse_agg.{c,h}`, `pg_aggregate.c` and all the `get_func_` functions in `lsyscache.c` for insight. For instance, `lookup_agg_function()` sounds useful (although it is static).